dali/operators/generic/transpose/transpose.h

```diff
@@ -16,37 +16,82 @@
 #define DALI_OPERATORS_GENERIC_TRANSPOSE_TRANSPOSE_H_
 
 #include <algorithm>
+#include <functional>
 #include <memory>
 #include <vector>
 
-#include "dali/pipeline/operator/operator.h"
+#include "dali/kernels/common/utils.h"
 #include "dali/operators/generic/transpose/cutt/cutt.h"
+#include "dali/pipeline/operator/operator.h"
 
 namespace dali {
 
 namespace detail {
 
-template <typename T>
-T Permute(const T& in, const std::vector<int>& permutation) {
-  T out = in;
-  for (size_t i = 0; i < permutation.size(); i++) {
-    auto idx = permutation[i];
-    out[i] = in[idx];
+template <typename ShapeT>
+inline void RowToColumnMajor(ShapeT *dims, int *perm, size_t len) {
+  std::reverse(dims, dims + len);
+  std::reverse(perm, perm + len);
+  for (size_t i = 0; i < len; ++i) perm[i] = len - 1 - perm[i];
+}
+
+// enough to represent batches of 4D
+constexpr int kStaticShapeElements = 5;
+using VecInt = SmallVector<int, kStaticShapeElements>;
```

Comment:
We're using `SmallVector<int64_t, 6>` in TensorView - I'd use value of 6 here, too.

Reply:
Done